geonode/harvesting/harvesters/wms.py

```diff
@@ -104,66 +136,179 @@ def check_availability(self, timeout_seconds: typing.Optional[int] = 5) -> bool:
-        params = self._base_wms_parameters
+            harvestable_resource: "HarvestableResource",  # noqa
+            harvesting_session_id: int
+    ) -> typing.Optional[base.HarvestedResourceInfo]:
+        resource_unique_identifier = harvestable_resource.unique_identifier
+        data = self._get_data()
+        for layer in data['layers']:
+            if resource_unique_identifier == layer['name']:
+                # WMS does not provide uuid, so needs to generated on the first time
+                # for update, use uuid from geonode resource
+                resource_uuid = uuid.uuid4()
+                if harvestable_resource.geonode_resource:
+                    resource_uuid = uuid.UUID(harvestable_resource.geonode_resource.uuid)
+                # WMS does not provide the date of the resource.
+                # Use current time for the date stamp and resource time.
+                time = datetime.now()
+                contact = resourcedescriptor.RecordDescriptionContact(**data['contact'])
+                return base.HarvestedResourceInfo(
+                    resource_descriptor=resourcedescriptor.RecordDescription(
+                        uuid=resource_uuid,
+                        point_of_contact=contact,
+                        author=contact,
+                        date_stamp=time,
+                        identification=resourcedescriptor.RecordIdentification(
+                            name=layer['name'],
+                            title=layer['title'],
+                            date=time,
+                            date_type='',
+                            originator=contact,
+                            graphic_overview_uri='',
+                            place_keywords=[],
+                            other_keywords=layer['keywords'],
+                            license=[],
+                            abstract=layer['abstract'],
+                            spatial_extent=layer['spatial_extent']
+                        ),
+                        distribution=resourcedescriptor.RecordDistribution(
+                            legend_url=layer['legend_url'],
+                            wms_url=layer['wms_url']
+                        ),
+                        reference_systems=layer['crs'],
+                    ),
+                    additional_information=None
+                )
+        return None
+
+    def _get_data(self) -> dict:
+        """ Return data from the harvester url in json format"""
+        params = self._base_wms_parameters.copy()
         params.update({
             "request": "GetCapabilities",
         })
         get_capabilities_response = self.http_session.get(
             self.remote_url, params=params)
         get_capabilities_response.raise_for_status()
+
         root = etree.fromstring(get_capabilities_response.content, parser=XML_PARSER)
         nsmap = _get_nsmap(root.nsmap)
-        useful_layers_elements = []
+
+        # get layers from element tree
```

Comment:
Please remove this type of comment from the code - it is not adding any value, since it is already clear what is going on below.

Comments are very useful to explain some context about the code or explain some harder to understand particularities. But if they are over used they end up being just visual noise.